### PR TITLE
Add a new flag --shake-profiling DIR

### PR DESCRIPTION
**Pull request by pepeiborra**
The flag provides a way to enable Shake profiling reports without recompiling.
Debug output prints links to the profiling artifacts for convenience.

### CODE REVIEW

File: src/Development/IDE/Core/Shake.hs [hunk 1]
```diff
@@ -392,9 +395,12 @@ shakeRun IdeState{shakeExtras=ShakeExtras{..}, ..} acts =
                   let res' = case res of
                           Left e -> "exception: " <> displayException e
                           Right _ -> "completed"
+                      profile = case res of
+                          Right (_, Just fp) -> ", profile saved at file://" <> fp
```
Comment (cocreature) [inline src/Development/IDE/Core/Shake.hs]: This isn’t going to produce a valid URI in general, if you want the URI (I guess to be able to click on it in VSCode?), you can use `filePathToUri'` or alternatively, just print out the filepath.

File: src/Development/IDE/Core/Shake.hs [hunk 2]
```diff
@@ -227,14 +227,17 @@ data IdeState = IdeState
...
-shakeRunDatabaseProfile :: Maybe FilePath -> ShakeDatabase -> [Action a] -> IO [a]
+shakeRunDatabaseProfile :: Maybe FilePath -> ShakeDatabase -> [Action a] -> IO ([a], Maybe FilePath)
 shakeRunDatabaseProfile mbProfileDir shakeDb acts = do
         (time, (res,_)) <- duration $ shakeRunDatabase shakeDb acts
-        whenJust mbProfileDir $ \dir -> do
-            count <- modifyVar profileCounter $ \x -> let !y = x+1 in return (y,y)
-            let file = "ide-" ++ profileStartTime ++ "-" ++ takeEnd 5 ("0000" ++ show count) ++ "-" ++ showDP 2 time <.> "html"
-            shakeProfileDatabase shakeDb $ dir </> file
-        return res
+        proFile <- case mbProfileDir of
```
Comment (cocreature) [inline src/Development/IDE/Core/Shake.hs]: this can be simplified a bit with `for` or `traverse`.

**Comment by pepeiborra** [comment]
Thanks for the feedback, I have applied it, rebased on master, and force-pushed.

**Comment by cocreature** [comment]
Thanks for addressing the comments!

**Pull request merged**